lib/compat/wordpress-6.7/blocks.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Temporary compatibility shims for block APIs present in Gutenberg.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Allow passing a PHP file as `variations` field for Core versions < 6.7
+ *
+ * @param array $settings Array of determined settings for registering a block type.
+ * @param array $metadata Metadata provided for registering a block type.
+ * @return array The block type settings
+ */
+function gutenberg_filter_block_type_metadata_settings_allow_variations_php_file( $settings, $metadata ) {
+	// If `variations` is a string, it's the name of a PHP file that
+	// generates the variations.
+	if ( ! empty( $settings['variations'] ) && is_string( $settings['variations'] ) ) {
+		$variations_path = wp_normalize_path(
+			realpath(
+				dirname( $metadata['file'] ) . '/' .
+				remove_block_asset_path_prefix( $metadata['variations'] )
```

Comment:
One thing is puzzling here to me. The condition checks the processed `settings` array, but the path gets computed using the `metadata` loaded from the `block.json` file. Why not use settings, too?

```suggestion
				remove_block_asset_path_prefix( $settings['variations'] )
```

Reply:
That's a fair point. I copied it over from https://github.com/WordPress/wordpress-develop/pull/6668 where I'm consistently using `$metadata` to denote the "input" and `$settings` for the output, in keeping with the structure of `register_block_type_from_metadata()`.

I missed this instance of when changing `$metadata['variations']` to `$settings['variations']` in the filter.